pkg/test/istio.io/examples/example.go

```diff
@@ -47,24 +51,31 @@ type Example struct {
 }
 
 // New returns an instance of an example test
-func New(t *testing.T, name string) Example {
-	return Example{
+func New(t *testing.T, name string) *Example {
+	return &Example{
 		name:  name,
 		steps: make([]testStep, 0),
 		t:     t,
 	}
 }
 
 // RunScript adds a directive to run a script
-func (example *Example) RunScript(script string, output outputType) {
-	//fullPath := getFullPath(istioPath + script)
-	example.steps = append(example.steps, newStepScript("./"+script, output))
+func (example *Example) RunScript(script string, output outputType, verifier VerificationFunction) *Example {
+	example.steps = append(example.steps, newStepScript(script, output, verifier))
+	return example
+}
+
+//WaitForPods adds a directive to wait for pods provided by a PodFunc to deploy
+func (example *Example) WaitForPods(fetchFunc KubePodFetchFunc) *Example {
+	example.steps = append(example.steps, newWaitForPodTestType(fetchFunc))
+	return example
 }
 
-// Apply applies an existing file
-func (example *Example) Apply(namespace string, path string) {
+// AddFile adds an existing file
+func (example *Example) Apply(namespace string, path string) *Example {
```

Comment:
What's the context the path is used in? That is, is this path relative to something specific?